contracts/pool/pools/liquid/LiquidPool.sol

```diff
@@ -0,0 +1,181 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.18;
+
+import "../TokenPoolBase.sol";
+import "./WithdrawalEligibility.sol";
+
+error ExistentTokensToClaim();
+
+/**
+ * @title LiquidPool
+ * @dev Provides functionality for staking, locking, and withdrawing tokens
+ * Tokens can be withdrawn with respecting grace period or immediately by paying ignore fee
+ */
+contract LiquidPool is TokenPoolBase, WithdrawalEligibility {
+    /**
+     * @notice Stakes tokens in the liquid pool
+     * @param amount The amount of tokens to stake
+     */
+    function liquidStake(
+        uint256 amount
+    ) external onlySufficientAmount(amount) nonReentrant {
+        Stake memory stakeInfo = stakes[msg.sender];
+
+        stakes[msg.sender] = Stake({
+            amount: stakeInfo.amount + amount,
+            eligbleWithdraw: stakeInfo.eligbleWithdraw
+        });
+
+        _claimTokens(amount);
+
+        emit TokenActivity(ActivityType.Stake, msg.sender, amount);
+    }
+
+    /**
+     * @notice Immediately withdraws staked tokens and pays grace period removal fee
+     * @param amount The amount of tokens to withdraw
+     */
+    function withdraw(
+        uint256 amount
+    ) external onlySufficientAmount(amount) nonReentrant {
+        Stake memory stakeInfo = stakes[msg.sender];
+
+        uint256 availableToWithdraw = stakeInfo.amount;
+
+        _checkAvailableWithdraw(availableToWithdraw, amount);
+
+        uint256 amountToWithdraw = _substractRemovalFee(amount);
+
+        stakeInfo.eligbleWithdraw = _initEligbleWithdraw();
+        stakeInfo.amount = stakeInfo.amount - amount;
+
+        stakes[msg.sender] = stakeInfo;
+
+        _releaseTokens(amountToWithdraw);
+
+        emit TokenActivity(ActivityType.Withdraw, msg.sender, amountToWithdraw);
+    }
+
+    /**
+     * @notice Withdraws staked tokens after starting grace period
+     * @param ignoreGracePeriod Whether to ignore the grace period after starting it
+     */
+    function withdraw(bool ignoreGracePeriod) external nonReentrant {
+        Stake memory stakeInfo = stakes[msg.sender];
+
+        uint256 availableToWithdraw = stakeInfo.eligbleWithdraw.amount;
+
+        _checkWithdrawEligibility(availableToWithdraw);
+
+        uint256 amountToWithdraw = _getAmountToWithdraw(
+            stakeInfo.eligbleWithdraw,
+            ignoreGracePeriod
+        );
```

Comment:
Sta se dogada kad je ignoreGracePeriod true? Onda user moze povuc tokene ali uz nekakav fee?

Reply:
Da da tako je. Moze kao ignorirat ovaj grace period od 4 tjedna i uzet tokene odma ali uz naknanadu.

Reply:
Treba li onda taj fee prebacit na neki endemic wallet?

Reply:
To ti nisan siguran stvarno. Neznan jel im cilj ovim fee-em utjecat na likvidnost token na tržištu ili kao penalizirat usera samo sta nezeli cekat grace period.